runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -1742,6 +1750,19 @@ J9::AheadOfTimeCompile::dumpRelocationData()
             }
             break;
 
+         case TR_MethodCallAddress:
```

Comment:
This should be added to `J9::AheadOfTimeCompile::dumpRelocationHeaderData` rather than here.

Reply:
Can you clarify why we have methods like `initializeCommonAOTRelocationHeader` and `dumpRelocationHeaderData` that manipulate the entire record, not just the header?

In the previous case I was writing to both a header field and the record's data portion, and in this case I'm dumping the record's data, not the header.

Reply:
I'm currently in the process of consolidating the way we write out the relocation header data to be consistent with the way we read out the relocation header data (https://github.com/eclipse/openj9/issues/3031). The latter is done via the `TR_RelocationRecord`s (and sub classes), so the former should also use the same interface. Eventually I want to clean up the way we use raw pointer manipulation in `dumpRelocationData` but first I wanted to consolidate everything before I started cleaning that up.

Reply:
I guess another reason for using the `TR_RelocationRecord`s is for consistency; plus, eventually @mstoodle  wants the the binary templates moved to `RelocationRecord.cpp` and so this function will not be able to use them anymore.

Reply:
Changes made.